app/controllers/landing_page_controller.rb

```diff
@@ -203,20 +220,48 @@ def render_landing_page(community_id:, default_locale:, locale_param:, structure
     )
 
     render_to_string :landing_page,
-           locals: { font_path: font_path,
+           locals: { font_path: FONT_PATH,
                      styles: landing_page_styles,
                      javascripts: {
-                       location_search: location_search_js
+                       location_search: location_search_js,
+                       translations: js_translations(locale_param, default_locale)
                      },
                      page: denormalizer.to_tree(structure, root: "page"),
-                     sections: denormalizer.to_tree(structure, root: "composition") }
+                     sections: denormalizer.to_tree(structure, root: "composition"),
+                     topbar_props: props,
+                     topbar_enabled: topbar_enabled }
   end
 
   def render_not_found(msg = "Not found")
     self.status = 404
     self.response_body = msg
   end
 
+  def topbar_props(community, community_customization, request_path, locale_param, default_locale)
+    # TopbarHelper pulls current lang from I18n
+    I18n.locale = locale_param.present? ? locale_param : default_locale
+
+    path =
+      if locale_param.present?
+        request_path.gsub(/^\/#{locale_param}/, "").gsub(/^\//, "")
+      else
+        request_path.gsub(/^\//, "")
+      end
+
+    TopbarHelper.topbar_props(
+      community: community,
+      path_after_locale_change: path,
+      search_placeholder: community_customization&.search_placeholder,
+      locale_param: locale_param)
+  end
+
+  def fetch_topbar_enabled(community_id)
+    flags_res = FeatureFlagService::API::Api.features.get(community_id: community_id)
+    flags_res.maybe
```

Comment:
Remember, `result.maybe` was kind of a "deprecated" method, because for errored results it returns `None`.